frontend/src/app/app.component.ts

```diff
@@ -62,6 +71,23 @@ export class AppComponent {
       const balanceStr = utils.formatEther(balanceBN);
       this.userEthBalance = parseFloat(balanceStr);
     });
+    this.displayTokenBalance(privateKey);
+  }
+
+  async displayTokenBalance(privateKey: string) {
+    console.log("Token address: ", this.lotteryTokenContractAddress);
+    if (!this.lotteryTokenContractAddress) return;
+    this.lotteryTokenContract = new Contract(
+      this.lotteryTokenContractAddress,
+      lotteryJson.abi,
+      this.userWallet ?? this.provider
+    );
+    const balanceBN = await this.lotteryTokenContract['balanceOf'](
+      this.userWallet
+    );
+
+    const balance = ethers.utils.formatEther(balanceBN);
+    this.userTokenBalance = parseFloat(balance);
   }
```

Comment:
And then here, calling the `displayTokenBalance` once a user connects their wallet, but it's failing on line 85 🤔